src/bokeh/command/subcommands/settings.py

```diff
@@ -99,6 +86,38 @@
     'Settings',
 )
 
+#-----------------------------------------------------------------------------
+# Helpers
+#-----------------------------------------------------------------------------
+
+@dataclass
+class ResolutionResult:
+    exact_matches: list[str] = field(default_factory=list)
+    fuzzy_matches: dict[str, list[str]] = field(default_factory=dict)
+    not_found: list[str] = field(default_factory=list)
+
+
+def resolve_setting_names(input_names: list[str], all_settings: dict[str, Any]) -> ResolutionResult:
+    """Resolve user-supplied setting names into matches against all_settings."""
+    result = ResolutionResult()
+
+    for name in input_names:
+        if name in all_settings:
+            result.exact_matches.append(name)
+            continue
+
+        substring_matches = [k for k in all_settings if name.lower() in k.lower()]
+        if substring_matches:
+            result.exact_matches.extend(substring_matches)
+        else:
+            close = get_close_matches(name, all_settings.keys(), n=3, cutoff=0.6)
+            if close:
+                result.fuzzy_matches[name] = close
+            else:
+                result.not_found.append(name)
+
+    return result
```

Comment:
If you are interested, `resolve_setting_names` could be put directly under unit test now, and it should be possible to test the behavior very comprehensively